### PR TITLE
Clare/issue#98/symbol set schema support

### PR DESCRIPTION
**Pull request by blaise17**
Don't explode if a symbol set is used for an image, just gracefully don't display it as we don't have them.

### CODE REVIEW

**Comment by joereddington** [comment]
Is there a handy 'failing' test file I can use to check?

**Comment by blaise17** [comment]
If you mean a test file that has a symbol set in, then yes. test/fixtures/obz/symbolset.obz
It is the test file used in the test I added to test this.

File: src/app/obfboard.ts [hunk 1]
```diff
@@ -300,6 +304,9 @@ export class OBFBoard {
       throw new FatalOpenVoiceFactoryError(ErrorCodes.OBF_VALIDATION, all_errors.join('\n'));
     }
 
+    // filter out images we can't display AFTER collecting error messages
+    this.images = this.images.filter(image => image.data || image.path || image.url);
```
Comment (psi77) [inline src/app/obfboard.ts]: Nice!

**Comment by joereddington-public** [comment]
Okay, tested with https://joereddington-public.github.io/testobz/symbolset.obz  - the behavour I got was "ignore new file and gracefully display the default file rather than the last sucessfully loaded" - do we want that behaviour? (I also feel like I want an error message, even as an alert, just telling the person in the config what's happening...)

**Comment by blaise17** [comment]
You're supposed to be getting the file you asked for but with no image for
the one we have no symbol set for...
Are there any errors in the console? Can it actually load the file from
that url?

**Comment by joereddington-public** [comment]
I am foolish. I hadn't noticed that the default file didn't have many images on it...  I will approve this.

**Pull request merged**